Refine ImportProcess enums and importing code

sethrj opened the pull request:
The `ImportX` enums have "none", "unknown", and "not_defined". This changes them all to `none` and marks the old name as deprecated. In preparation for some MSC work I've moved the PhysicsTable conversion to its own helper function. Finally, I've removed `G4SystemOfUnits.h` to avoid importing CLHEP units into the global namespace: they're now explicitly included in the files that need them with a `using` directive.

stognini commented on src/celeritas/ext/detail/ImportProcessConverter.cc:
```diff
@@ -148,68 +150,11 @@ ImportModelClass to_import_model(G4VEmModel const& model)
-{
-    switch (g4_vector_type)
-    {
-#if CELERITAS_G4_V10
-        case T_G4PhysicsVector:
-            return ImportPhysicsVectorType::unknown;
-#endif
-        case T_G4PhysicsLinearVector:
-            return ImportPhysicsVectorType::linear;
-        case T_G4PhysicsLogVector:
-#if CELERITAS_G4_V10
-        case T_G4PhysicsLnVector:
-#endif
-            return ImportPhysicsVectorType::log;
-        case T_G4PhysicsFreeVector:
-#if CELERITAS_G4_V10
-        case T_G4PhysicsOrderedFreeVector:
-        case T_G4LPhysicsFreeVector:
-#endif
```
I see why this is organized this way, but it might be easier to read (at the cost of more lines) if we had
```cpp
#if CELERITAS_G4_V10
    [G4 v10 cases]
#else
    [G4 v11 cases]
#endif
```

sethrj replied:
Even better, how about we drop support for removed V10 enums?

sethrj replied:
Note that I only *moved* the lines in this PR: this implementation was already in place. How about we plan a separate PR to remove support for Geant4@10-only features?

stognini replied:
That sounds good to me.

stognini commented:
Since now we have a `GeantImportUtils`, it might make sense to also move the `ImportProcessConverter::initialize_micro_xs_physics_vector(...)` to it.

sethrj commented:
`initialize_micro_xs_physics_vector` could actually be an anonymous namespace function since it's only used in that translation unit. I moved the table creator because I was going to use it from a separate MSC process converter. But now that I've implemented it honestly I think I could move it back...

stognini commented on src/celeritas/io/ImportPhysicsTable.hh:
```diff
@@ -65,10 +67,15 @@ enum class ImportUnits
 struct ImportPhysicsTable
 {
-    ImportTableType table_type;
+    ImportTableType table_type{ImportTableType::size_};
```
Why are we initializing this as a `size_`?

sethrj replied:
Since `size_` is always a placeholder meaning "past the last valid enumeration value" I often use it as a sentinel value for "unassigned", since most `switch` statements, `to_cstring` etc. will raise an error when that value is hit.

You do make me think that the `::none` models should be `::other` if they're valid but just not one of the enumerated types (which for now we can't support)? Maybe just delete those entries entirely and change to `::size_`? But that will be a breaking change because it'll invalidate our ROOT files by changing the enumeration...

stognini replied:
Ohh, good point about returning an error. I like having `other` for valid but not implemented and `size_` as past the last valid enum as a way of saying that we hit an error.
We can defer that to a future PR if you don't want to update the files just yet. Which makes sense, since that will have to be done anyway when the MSC import bit comes into play...

stognini commented:
Overall the PR looks good, with one last comment: We have set `CMAKE_CXX_STANDARD = 14`, and now I am hit with this:
```
/src/celeritas/io/ImportProcess.hh:56:13: warning: attributes on an enumerator declaration are a C++17 extension [-Wc++17-extensions]
    unknown [[deprecated]] = none,
```

sethrj commented:
Aha, I didn't realize that [attributes for enumerator values](https://www.cppstories.com/2017/07/cpp17-in-details-attributes/#attributes-for-namespaces-and-enumerators) was a C++17 extension. I'll remove it and add to the list of #421. Thanks!

stognini commented on src/celeritas/io/ImportProcess.hh:
```diff
@@ -87,9 +83,7 @@ enum class ImportProcessClass
-#ifndef SWIG
-    unknown [[deprecated]] = none,
-#endif
+    unknown CELER_DEPRECATED_POST = none,
```
Nice solution with the macro.

stognini commented:
Looks great. 🎉
